OpenRA.Game/CPos.cs

```diff
@@ -58,7 +58,13 @@ public CPos(int x, int y, byte layer)
 		public bool Equals(CPos other) { return Bits == other.Bits; }
 		public override bool Equals(object obj) { return obj is CPos && Equals((CPos)obj); }
 
-		public override string ToString() { return X + "," + Y; }
+		public override string ToString()
+		{
+			if (Layer == 0)
+				return X + "," + Y;
+
+			return X + "," + Y + "," + Layer;
```

Comment:
We need to be careful here - I don't see any special cases in `FieldSaver.FormatValue` for `CPos`, so I think (but haven't checked) that this will be used when serializing yaml trees.

The change here is a bugfix, but `FieldLoader.ParseCPos` still doesn't understand layers so will break on parsing!

Reply:
Agreed. Have added handling in the loader to accommodate.

It may well be worth retiring the `CPos(int, int)` ctor and forcing callsites to handle the layer explicitly - might help shake loose any remaining bugs due to layer ignorance.